From 83924426b3da6d2e366452c143b6819fa6b0f41e Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Mon, 12 May 2014 15:50:08 +0200
Subject: [PATCH 4/4] Fix run with Minitest 5

There's a warning due to the weird way we avoid autorun, but at least it works
with both minitest 4 (present in enterprise linux 7) and minitest 5 (shipped
with fedora 20):

Warning: you should require 'minitest/autorun' instead.
Warning: or add 'gem "minitest"' before 'require "minitest/autorun"'
---
 devel/levitate.rb           | 8 +++++++-
 test/comp_tree_test_base.rb | 1 +
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/devel/levitate.rb b/devel/levitate.rb
index 745b6e6..e425fcf 100644
--- a/devel/levitate.rb
+++ b/devel/levitate.rb
@@ -338,7 +338,13 @@ class Levitate
         test_files.each { |file| require file }
 
         # if we use at_exit hook instead, it won't run before :release
-        MiniTest::Unit.new.run ARGV
+        if MiniTest::Unit.method_defined? :run
+          # minitest 4
+          MiniTest::Unit.new.run ARGV
+        else
+          # minitest 5
+          Minitest.run ARGV
+        end
       end
       
       desc "run tests with coverage"
diff --git a/test/comp_tree_test_base.rb b/test/comp_tree_test_base.rb
index 2affe6f..55a0fa9 100644
--- a/test/comp_tree_test_base.rb
+++ b/test/comp_tree_test_base.rb
@@ -1,6 +1,7 @@
 $LOAD_PATH.unshift File.dirname(__FILE__) + '/../lib'
 $LOAD_PATH.unshift File.dirname(__FILE__) + '/../devel'
 
+gem 'minitest'
 require 'minitest/unit'
 require 'minitest/autorun' unless defined? Rake
 require 'comp_tree'
-- 
1.8.3.1

